gbpservice/neutron/services/servicechain/plugins/ncp/node_drivers/nfp_node_driver.py

```diff
@@ -576,11 +638,21 @@ def _get_service_targets(self, context):
         # Bug with NCP. For create, its not setting service targets in context
         if not service_targets:
             service_targets = context.get_service_targets(update=True)
-        for service_target in service_targets:
-            if service_target.relationship == 'consumer':
-                consumer_service_targets.append(service_target)
-            elif service_target.relationship == 'provider':
-                provider_service_targets.append(service_target)
+        if (not service_targets and service_type in
+            [pconst.FIREWALL, pconst.VPN]):
```

Comment:
According to get_plumbing_info, service targets are created only for FW right. So, here service type is VPN alone. I think no need to check for FW.